#!/usr/bin/env python
PACKAGE = "stroll_bearnav"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("showAllMatches",       	bool_t,    0,    "Show images with all matches", True)
gen.add("showGoodMatches",       	bool_t,    0,    "Show images with good matches", True)
gen.add("velocityGain",       		double_t,    0,    "Velocity gain", 1,     0, 5)
gen.add("matchingRatio",       		double_t,    0,    "Matching ratio of matcher", 0.7,     0, 1)
gen.add("maxVerticalDifference",        double_t,    0,    "Maximal vertical difference", 50,     0, 500)
gen.add("minGoodFeatures",        	int_t,    0,    "Minimal number of good features", 3,     0, 20)
gen.add("pixelTurnGain",        	double_t,    0,    "Pixel displacement to angular speed gain", 0.001,     0, 0.01)
gen.add("adaptiveSpeedMin",        	double_t,    0,    "Minimal adaptive speed", 1.0,     0, 5.0)
gen.add("adaptiveSpeedMax",        	double_t,    0,    "Maximal adaptive speed", 1.0,     0, 5.0)
gen.add("remapRatio",        		double_t,    0,    "Ratio of remapped features", 0.5,     0, 1.0)
gen.add("plasticMap",        		bool_t,    0,    "Exchange all features", False)
exit(gen.generate(PACKAGE, "stroll_bearnav", "navigator"))
